sync/src/tasks/block_sync_task.rs

```diff
@@ -682,85 +646,12 @@ where
                 })?;
                 parent_block.children.push(block_header.id());
                 self.local_store.save_dag_sync_block(parent_block)?;
```

Comment:
I find that save_dag_sync_block this function have called two times,  DagSyncBlock  init  with block 1 without children hashes, another is init block 1 with children hashs. It makes confusion.

Reply:
Because when it fetches a dag block, it cannot know its children until, in the later loop, the dag blocks are executed one by one and checked the relationship, attaching the their parents.